source/Patches/CrewmateRoles/HunterMod/CantReport.cs

```diff
@@ -0,0 +1,70 @@
+using System.Linq;
+using HarmonyLib;
+using TownOfUs.CrewmateRoles.MedicMod;
+using UnityEngine;
+
+namespace TownOfUs.CrewmateRoles.HunterMod
+{
+    [HarmonyPatch(typeof(PlayerControl), nameof(PlayerControl.FixedUpdate))]
+    public class CantReport
+    {
+        public static void Postfix(PlayerControl __instance)
+        {
+            if (!__instance.AmOwner) return;
+            if (!__instance.CanMove) return;
+            if (!__instance.Is(RoleEnum.Hunter)) return;
+            if (CustomGameOptions.HunterBodyReport) return;
+            var truePosition = __instance.GetTruePosition();
+
+            var data = __instance.Data;
+            var stuff = Physics2D.OverlapCircleAll(truePosition, __instance.MaxReportDistance, Constants.Usables);
+            var flag = (GameOptionsManager.Instance.currentNormalGameOptions.GhostsDoTasks || !data.IsDead) &&
+                       (!AmongUsClient.Instance || !AmongUsClient.Instance.IsGameOver) && __instance.CanMove;
+            var flag2 = false;
+
+            foreach (var collider2D in stuff)
+                if (flag && !data.IsDead && !flag2 && collider2D.tag == "DeadBody")
+                {
+                    var component = collider2D.GetComponent<DeadBody>();
+
+                    if (Vector2.Distance(truePosition, component.TruePosition) <= __instance.MaxReportDistance)
+                    {
+                        var matches = Murder.KilledPlayers.FirstOrDefault(x => x.PlayerId == component.ParentId);
+                        if (matches != null && matches.KillerId != PlayerControl.LocalPlayer.PlayerId) { 
+                            if (!PhysicsHelpers.AnythingBetween(__instance.Collider, truePosition, component.TruePosition, Constants.ShipOnlyMask, false)) flag2 = true; 
+                        }
+                    }
+                }
+
+            DestroyableSingleton<HudManager>.Instance.ReportButton.SetActive(flag2);
```

Comment:
the hunter will still be able to report via keybinds, maybe should save this bool as a field and use it in a prefix for doclick?

Reply:
Do the keybinds work on buttons that are inactive? I would think that's more of an issue with the keybinds.

Reply:
actually hmm i think i might have not meant keybinds, not sure if those still work but i meant like clicking the body

Reply:
I wasn't aware you could click on the body to report it. Maybe that's a mod? The Hunter just uses the Sheriff's report button restriction.